climada/util/coordinates.py

```diff
@@ -1059,10 +1147,13 @@ def match_coordinates(
     if np.array_equal(coords, coords_to_assign):
         assigned_idx = np.arange(coords.shape[0])
     else:
+        if threshold is None:
+            threshold = 2 * max(abs(r) for r in get_resolution(coords_to_assign.T))
```

Comment:
`get_1d_resolution` can be extremely costly. It takes unique coordinates and computes differences between them. Shouldn't we raise a warning that this is suboptimal?

https://github.com/CLIMADA-project/climada_python/blob/08ec51bb958436cd18e77cf8f4178bc05a295d78/climada/util/coordinates.py#L1836

Reply:
Testing, for a regular grid with 65'000 points (world at 1° resolution), it takes 13ms, for 6'500'000 points (world at 0.1° resolution) it takes 1.5s. Is this acceptable?

<img width="603" height="411" alt="Screenshot 2025-08-27 at 15 07 09" src="https://github.com/user-attachments/assets/9be941a9-2aa2-4065-93ef-7731c4f1767b" />

Reply:
Yes, that seems acceptable to me! Thanks a lot for the testing 🙌